src/backend/cdb/cdbtmpdir.c

```diff
@@ -17,113 +17,288 @@
  * under the License.
```

Comment:
looks the "failed_tmpdir" column in gp_segment_configuration does not contain the host name of the failed dir. It would be nice to have host name included to ease the failure diagnosis.